### PR TITLE
Add support for reading custom Kotlin compiler arguments

### PR DESCRIPTION
**Pull request by hzsweers**
This builds on #775 and adds support for reading compiler configuration directly from the kotlin compile task. Not written tests yet because I want to wait for #775 to get in, but it's a start and want to get this up now for early feedback.

This supports all the standard arguments configurable by the gradle plugin, as well as the important free arguments option to add new ones

### CODE REVIEW

File: buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java [hunk 1]
```diff
@@ -281,6 +285,46 @@ public static String javaVersion(JavaVersion version) {
       return ImmutableList.of();
     }
     ImmutableList.Builder<String> optionBuilder = ImmutableList.builder();
+
+    // Add all compiler arguments
+    String compileTaskName = isTest ? "compileTestKotlin" : "compileKotlin";
```
Comment (kageiit) [inline buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java]: Is there no public API/static constants for these task names? Hardcoding can lead to breakages in future

Comment (kageiit) [reply buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java]: This approach also won't work for kotlin Android targets

Comment (hzsweers) [reply buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java]: No public api/constants. Why won't this work for Android targets?

Comment (kageiit) [reply buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java]: Android targets don't have the same task names as they are variant aware

Comment (hzsweers) [reply buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java]: Good point. I can do something like `tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {`, but how can I connect those back to the specific variant at hand? Or would it just be basic String matching?

Comment (hzsweers) [reply buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java]: alternatively - just grab the first one and only support configurations that apply to all

Comment (thalescm) [reply buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java]: I tend to think people use flavors and buildTypes for feature specific things (changing a hardcoded string value, changing themes, etc...). Could not think of a use case where anyone would want to consistently compile differently depending on that, so I think it's valid to only support configurations that apply to all.

Comment (hzsweers) [reply buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java]: Agreed, just updated to take the first matching one in 245e32f

Comment (hzsweers) [reply buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java]: Did keep the test vs non-test distinction though. Ultimately it's a dynamic naming strategy though

```kotlin
    override val compileKotlinTaskName: String
        get() = lowerCamelCaseName(
            "compile",
            compilationName.takeIf { it != KotlinCompilation.MAIN_COMPILATION_NAME },
            "Kotlin",
            target.targetName
        )
```

**Comment by hzsweers** [comment]
Will do

**Pull request closed without merging**